# Nyest corpus statistics

In [8]:
import pandas as pd
df = pd.read_csv("../data/interim/cleaned_articles.csv", sep=";")
print(df.shape) # no of articles
df.head()

(18649, 3)


Title  \
0                                         Urál‒Altaj   
1                                Elhunyt Nagy Emília   
2                                Elhunyt Nagy Emília   
3  Brit utazók Magyarországon ‒hírek egy különös ...   
4  Brit utazók Magyarországon ‒hírek egy különös ...   

                                                Lead  \
0  Magyar őstörténeti tanulmányokban gyakran előf...   
1  Elhunyt Nagy Emília, az ELTE Finnugor Tanszéké...   
2  Elhunyt Nagy Emília, az ELTE Finnugor Tanszéké...   
3  A Kárpát-medencét felkereső angol és skót utaz...   
4  A Kárpát-medencét felkereső utazók között ango...   

                                             Article  
0   A helyesírási szabályzat 12. kiadása ‒ miként...  
1  https://www.nyest.hu/renhirek/elhunyt-nagy-emiila  
2   2022. július 31-én elhunyt Nagy Emília, az EL...  
3   Buda városának többnyelvű népességéről Charle...  
4   Korábban már írtunk az angol utazók északi fe...

In [10]:
import itertools
import spacy
from nltk.tokenize import sent_tokenize, word_tokenize

nlp = spacy.load("hu_core_news_lg")

characters = []
words = []
lemmas = []
sentences = []
for _, row in df.iterrows():
    characters.append(len(row["Article"].lower()))
    sents = sent_tokenize(row["Article"].lower())
    sentences.append(len(sents))
    wds = list(itertools.chain(*[word_tokenize(sent) for sent in sents]))
    words.append(len(wds))
    doc = nlp(row["Article"])
    ls = []
    for s in doc.sents:
        for wd in s:
            if wd.pos_ not in ["PUNCT", "X"]:
                ls.append(wd.lemma_.lower())
    lemmas.append(ls)
avg_chars = sum(characters) / len(characters)
avg_sents = sum(sentences) / len(sentences)
avg_words = sum(words) / len(words)
print(avg_chars, avg_sents, avg_words) # average number of characters, sentences and words (token) per article

3411.697678159687 24.694514451177007 547.5187409512574


In [15]:
lemmas_per_articel = [len(set(l)) for l in lemmas]
lexical_diversity_per_article = [len(set(l))/len(l) if len(l) > 0  else 0 for l in lemmas]

avg_lemmas = sum(lemmas_per_articel) / len(lemmas_per_articel)
avg_lexical_diversity = sum(lexical_diversity_per_article) / len(lexical_diversity_per_article)
print(avg_lemmas, avg_lexical_diversity) # average number of lemmas, average lexical diversity

247.13448442275725 0.6102234655801655


## Corpus statistics

- 18649 articles
- An average article contains 3411.697678159687 characters
- An average article contains 547.5187409512574 words (tokens)
- An average article contains 247.13448442275725 unique words (types)
- An average article contains 24.694514451177007 sentences
- The average lexical diversity of an article is 0.6102234655801655